### PR TITLE
[Merged by Bors] - chore: rename `by_contra'` to `by_contra!`

### PR DESCRIPTION
**Pull request by kim-em**
To fit with the "please try harder" convention of `!` tactics.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by YaelDillies — APPROVED** [review]
Thanks! It always surprised me this wasn't the name in the first place.

maintainer merge

**Comment by sgouezel** [comment]
bors d+
It will conflict with #8798, so please wait until this one is merged and merge master and fix before sending it to bors.

**Pull request closed without merging**
